course/latex/converter.py

```diff
@@ -0,0 +1,646 @@
+            from distutils.version import LooseVersion as LV
+            if LV(version) < LV(self.required_version):
+                error = Critical(
+                    "Version outdated",
+                    hint=("'%(tool)s' with version "
+                          ">=%(required)s is required, "
+                          "current version is %(version)s"
+                          )
+                         % {"tool": self.name,
+                            "required": self.required_version,
+                            "version": version
+                            },
+                    obj=self.name
+                )
+        return error
+
+
+class TexCompilerBase(CommandBase):
+    def __init__(self):
+        self.bin_path_dir = getattr(
+            settings, "RELATE_%s_BIN_DIR" % self.name.upper(),
+            getattr(settings, "RELATE_LATEX_BIN_DIR", "")
+        )
+        self.bin_path = os.path.join(
+            self.bin_path_dir, self.cmd.lower())
+
+
+class Latexmk(TexCompilerBase):
+    name = "latexmk"
+    cmd = "latexmk"
+    required_version = "4.39"
+
+
+class LatexCompiler(TexCompilerBase):
+    latexmk_option = (
+        '-latexoption="-no-shell-escape '
+        '-interaction=batchmode -halt-on-error "'
+    )
+
+    @property
+    def output_format(self):
+        raise NotImplementedError()
+
+    def __init__(self):
+        super(LatexCompiler, self).__init__()
+        self.latexmk_prog_repl = self._get_latexmk_prog_repl()
+
+    def _get_latexmk_prog_repl(self):
+        """
+        Program replace when using "-pdflatex=" or "-latex="
+        arg in latexmk, especially needed when compilers are
+        not in system's default $PATH.
+        :return: the latexmk arg "-pdflatex=/path/to/pdflatex" for
+        # pdflatex or "-pdflatex=/path/to/xelatex" for xelatex
+        """
+        return (
+            "-%s=%s" % (self.name.lower(), self.bin_path.lower())
+        )
+
+    def get_latexmk_subpro_cmdline(self, input_path):
+        latexmk = Latexmk()
+        return [
+            latexmk.bin_path,
+            "-%s" % self.output_format,
+            self.latexmk_prog_repl,
+            self.latexmk_option,
+            input_path
+        ]
+
+
+class Latex(LatexCompiler):
+    name = "latex"
+    cmd = "latex"
+    output_format = "dvi"
+
+
+class PdfLatex(LatexCompiler):
+    name = "PdfLatex"
+    cmd = "pdflatex"
+    output_format = "pdf"
+
+
+class XeLatex(LatexCompiler):
+    name = "XeLatex"
+    cmd = "xelatex"
+    output_format = "pdf"
+    def __init__(self):
+        super(XeLatex, self).__init__()
+        self.latexmk_prog_repl = (
+            "-%s=%s" % ("pdflatex", self.bin_path)
+        )
+
+
+class Imageconverter(CommandBase):
+
+    @property
+    def output_format(self):
+        raise NotImplementedError
+
+    def __init__(self):
+        bin_path_dir = getattr(
+            settings, "RELATE_%s_BIN_DIR" % self.name.upper(),
```

Comment:
Yes, it's [here](https://github.com/inducer/relate/pull/207/files/6d2d8f5032874af4f1a26017e3c175ce413f45f8#diff-fa1395737ba4b4fa8329b63aff5be72dR154). Or, did you mean besides ``ImageMagick``, path to ``dvisvgm`` and ``dvipng`` should also be configured?